azure-iot-hub-devicesdk/azure/iot/hub/devicesdk/message.py

```diff
@@ -0,0 +1,53 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+import logging
+
+
+logger = logging.getLogger(__name__)
+
+
+class Message(object):
+    def __init__(
+        self, data, creation_time_utc, message_id=None, content_encoding=None, content_type=None
+    ):
+        """
+        :param data: The  data that constitutes the payload
+        :param creation_time_utc: Date and time the message was created
+        :param message_id: A user-settable identifier for the message used for request-reply patterns.Format: A case-sensitive string (up to 128 characters long) of ASCII 7-bit alphanumeric characters + {'-', ':', '.', '+', '%', '_', '#', '*', '?', '!', '(', ')', ',', '=', '@', ';', '$', '''}.
+        :param content_encoding: Content encoding of the message data. can be 'utf-8', 'utf-16' or 'utf-32'.
+        :param content_type: Content type property used to routes with the message body. Should be 'application/json'.
+        """
+        self.data = data
+        self.creation_time_utc = creation_time_utc
+        self.custom_properties = {}
+        self.lock_token = None
+        self.message_id = message_id
+        self.sequence_number = None
+        self.to = None
+        self.expiry_time_utc = None
+        self.enqueued_time = None
+        self.correlation_id = None
+        self.user_id = None
+        self.ack = None
+        self.content_encoding = content_encoding
+        self.content_type = content_type
+        self.device_id = None
+        self.auth_generation_id = None
+        self.auth_method = None
+
+    def add_property(self, key, value):
+        """
+        Adds the key-value pair to the collection. The strings 'Authorization'
+        and 'iothub-to' are considered as reserved key names and will cause the
+        function to log error.
+        """
+        if key == "Authorization" or key == "iothub-to":  # Not sure of this. Was part of Node code.
+            logger.error(
+                "Cannot add these keys"
+            )  # Should we throw exception ? Should we return false ?
```

Comment:
># Should we throw exception ? Should we return false ? [](start = 15, length = 54)

always throw on failure unless you have a really good reason.

Reply:
I believe Authorization is forbidden only in HTTP and iothub-to actually allowed on the service side, that could share the Message class. I believe it's premature to add those checks.

---
In reply to: [237661027](https://github.com/Azure/azure-iot-sdk-python-preview/pull/27#discussion_r237661027) [](ancestors = 237661027)

Reply:
The exception as there already, Should have removed the comment

---
In reply to: [237661027](https://github.com/Azure/azure-iot-sdk-python-preview/pull/27#discussion_r237661027) [](ancestors = 237661027)